JamfSync/Model/DistributionPoint.swift

```diff
@@ -383,13 +313,108 @@ class DistributionPoint: Identifiable {
 
     // MARK: - Private functions
 
+    private func copyFilesToDst(sourceName: String, willDownloadFiles: Bool, filesToSync: [DpFile], dstDp: DistributionPoint, jamfProInstance: JamfProInstance?, forceSync: Bool, progress: SynchronizationProgress) async throws {
+        isCanceled = false
+        filesWereZipped = false
+        var someFileSucceeded = false
+        var someFilesFailed = false
+        var downloadMultiple: Int64 = 1
+        if willDownloadFiles {
+            downloadMultiple = 2
+        }
+        progress.totalSize = calculateTotalTransferSize(filesToSync: filesToSync) * downloadMultiple
+        var currentTotalSizeTransferred: Int64 = 0
+        var lastFile: DpFile?
+        var lastFileTansferred = false
+        inProgressDstDp = dstDp
+        for dpFile in filesToSync {
+            lastFile = dpFile
+            progress.initializeFileTransferInfoForFile(operation: "Copying", currentFile: dpFile, currentTotalSizeTransferred: currentTotalSizeTransferred)
+
+            do {
+                lastFileTansferred = false
+                var localFileUrl: URL?
+                if isCanceled { break }
+                if willDownloadFiles {
+                    Task { @MainActor in
+                        progress.operation = "Downloading"
```

Comment:
you should be able to `await` these `MainActor` isolated properties instead of surrounding them in a `Task` block. Like so: `await progress.operation = "Downloading"`

Reply:
That gives me "No 'async' operations occur within 'await' expression.

Reply:
Oh i see. It's not main actor isolated. Just `@Published` and needs to be modified on the main thread